src/dvc_data/hashfile/transfer.py

```diff
@@ -224,9 +224,6 @@ def transfer(
     if not status.new:
         return TransferResult(set(), set())
 
-    if callback != DEFAULT_CALLBACK:
-        callback = callback.as_tqdm_callback(unit="file", desc="Transferring")
```

Comment:
dvc-data HashFileDB shouldn't be generating its own progress bars, DVC should be explicit and generate progress bars when it wants them.

I think at some point there was a default argument change in DVC, where `dvc push` assumed progress bars would come from dvc-data, and once that was dropped we ended up with no progress at all on `dvc push`.

Reply:
Alternatively, we could also set the `callback` kwarg to `None` and default to `tqdm_callback` here.

Reply:
I think the only time we should be generating a progress bar at all in dvc-data or dvc-objects is if we want to wrap/branch a nested progress level (where we generate a nested byte level bar for a particular file operation when dvc has passed us a per-file-level bar for a batch operation)

Reply:
Agreed, it's just that we call `transfer` in multiple places, so we have to remember to pass the callback.

Reply:
@skshetry yes, this is being done on the DVC side (https://github.com/iterative/dvc/pull/9793)

Reply:
> Alternatively, we could also set the callback kwarg to None and default to tqdm_callback here.

I didn't dig into the git history but iirc at some point it was defaulted to `None`, and this is where the `dvc push` pbar came from.

I think what probably happened is that the default arg was changed from `None` to `DEFAULT_CALLBACK` to fix type checking errors (since callback is typed `Callback` and not `Optional[Callback]` everywhere in dvc-data/dvc-objects/fsspec). Whenever that change was made we did not re-add progress bars in DVC, so `dvc push` stopped reporting progress.

Reply:
There are similar progress bar callbacks all over dvc-data. Is this a specific case where it should be moved to dvc, or should we be moving them all?

Reply:
@dberenbaum most of them should be moved to DVC, but some of them do belong here because they are specifically for dvc-data level nested progress bars (where DVC displays progress for total files and then dvc-data or dvc-objects adds an additional level for bytes within the files currently being transferred)